src/pages/index.js

```diff
@@ -62,27 +113,130 @@ const editProfileModal = new PopupWithForm(
 );
 editProfileModal.setEventListeners();
 
+const editAvatarPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  handleAvatarEditSubmit
+);
+editAvatarPopup.setEventListeners();
+
+const deleteCardPopup = new PopupWithConfirmation("#confirm-delete-modal");
+deleteCardPopup.setEventListeners();
+
 function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", openImagePreview);
-  return cardElement.getView();
+  const cardElement = new Card(
+    cardData,
+    "#card-template",
+    openImagePreview,
+    handleLikeCard,
+    handleDeleteCard
+  );
+  const view = cardElement.getView();
+  return view;
+}
+
+function handleDeleteCard(card) {
+  deleteCardPopup.setAction(() => {
+    api
+      .deleteCard(card.getCardId())
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => console.error("Error deleting card:", err));
+  });
+  deleteCardPopup.open(card);
+}
+
+function handleLikeCard(card) {
+  if (card.isLiked) {
+    return api
+      .removeLike(card.getCardId())
+      .then(() => {
+        card.setCardLikes(false);
+      })
+      .catch((err) => console.error("Failed to dislike card:", err));
+  } else {
+    return api
+      .addLike(card.getCardId())
+      .then(() => {
+        card.setCardLikes(true);
+      })
+      .catch(console.error);
+  }
 }
 
 function handleProfileEditSubmit(formValues) {
-  userInfo.setUserInfo({
+  const apiData = {
     name: formValues.title,
     about: formValues.description,
-  });
-  editProfileModal.close();
+  };
+
+  editProfileModal.renderLoading(true);
+  api
+    .setUserInfo(apiData)
+    .then((data) => {
+      const currentUserInfo = userInfo.getUserInfo();
+
+      userInfo.setUserInfo({
+        name: data.name,
+        about: data.about,
+        avatar: currentUserInfo.avatar,
+      });
+      profileEditForm.reset();
+      profileEditValidator.resetValidation();
+      editProfileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(() => {
+      editProfileModal.renderLoading(false);
+    });
 }
 
 function handleAddCardFormSubmit(formValues) {
-  const name = formValues.title;
-  const link = formValues.link;
+  const cardData = { name: formValues.title, link: formValues.link };
+
+  addCardModal.renderLoading(true);
+  api
+    .addCard(cardData)
+    .then((newCard) => {
+      const cardElement = createCard(newCard);
+      cardList.addItem(cardElement);
+      addCardFormElement.reset();
+      addCardFormValidator.resetValidation();
```

Comment:
`      addCardFormValidator.resetValidation();
`

Please, delete it

`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submitting.